### ISSUE TITLE
Can't access CBB (CBC) binary using flopy binary reader class

### ISSUE THREAD

**Issue by JFullerton-intera** [modflowpy/flopy#703]
I am unable to read any portion of the CBB file using the binaryfile class. See below:
`import flopy.utils.binaryfile as bf`
`cbb = bf.CellBudgetFile(r"P2Rv8.2_1000yr.cbb")`

This results in the following error:

> Traceback (most recent call last):
>   File "<input>", line 1, in <module>
>   File "C:\ProgramData\Miniconda3\envs\ca_cie\lib\site-packages\flopy\utils\binaryfile.py", line 618, in __init__
>     self._build_index()
>   File "C:\ProgramData\Miniconda3\envs\ca_cie\lib\site-packages\flopy\utils\binaryfile.py", line 708, in _build_index
>     self._skip_record(header)
>   File "C:\ProgramData\Miniconda3\envs\ca_cie\lib\site-packages\flopy\utils\binaryfile.py", line 768, in _skip_record
>     self.file.seek(nbytes, 1)
> OSError: [Errno 22] Invalid argument

Is there a specific way that we need to export our CBB file (e.g. "COMPACT")?

I have verified that I can use the Windows executable of the zone budget tool supported/created by USGS. In other words, I can generate zone budget output and read the CBB file using the USGS executable documented [here](https://water.usgs.gov/software/code/ground_water/zonebdgt/doc/ofr90392.pdf).

**Comment by JFullerton-intera** [comment]
This file is larger than 2 GB, is there a link to how this problem was resolved (if it has been)? Otherwise I can provide the CBB file and we can work through that for this issue. How do we proceed?

**Comment by langevin-usgs** [comment]
Are you using a 32-bit computer or a 32-bit version of Python?  I believe that was the cause of previous problems with large budget files.

**Comment by JFullerton-intera** [comment]
I’m on a 64-bit system and interpreter (see console startup message):

“Python 3.6.9 |Anaconda, Inc.| (default, Jul 30 2019, 14:00:49) [MSC v.1915 64 bit (AMD64)] on win32”

**Comment by langevin-usgs** [comment]
Try setting verbose=True in the budget file constructor and see if that provides any hints.

**Comment by JFullerton-intera** [comment]
Ha! I found the problem. The ‘verbose=True’ didn’t end up giving me the answer (returned the same error). I was digging deeper and I realized that the flopy.utils.zonbud.ZoneBudget() class applies the default options as used in the flopy.utils.binaryfile.CellBudgetFile() class. My CBB is in double precision, so, to fix this workflow for me, I used the following sequence:

import flopy
import flopy.utils.binaryfile as bf

zon_dict = flopy.utils.zonbud.read_zbarray(‘zon_file’)
cbb = bf.CellBudgetFile(‘P2Rv8.2.cbb’, precision=’double’)
zb = flopy.utils.zonbud.ZoneBudget(cbb, zon_dict, verbose=True)  # verbose option not needed

That worked like a charm.
